src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.Helpers.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics.Arm;
+using System.Runtime.Intrinsics.X86;
+
+namespace System.Runtime.Intrinsics
+{
+    // Contains internal helper methods
+    public static partial class Vector128
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static Vector128<T> ConditionalSelectBitwise<T>(Vector128<T> selector, Vector128<T> ifTrue, Vector128<T> ifFalse)
+            where T : struct
+        {
+            Debug.Assert(Sse.IsSupported || AdvSimd.IsSupported);
+
+            if (Sse.IsSupported)
+            {
+                ifTrue = Sse.And(ifTrue.AsSingle(), selector.AsSingle());
```

Comment:
Would it better to pass this directly to `AndNot`? Reusing variables like this makes it harder for the JIT to optimize the code.

Reply:
More importantly that isn't valid, i messed up generic-fying the code 💯 , fixed now. I checked codegen for having locals vs directly pasisng in, and there is no difference

Reply:
There may be a difference once you use this in a real-world code and start hitting JIT limits on too complex code - that is fairly easy with multiple layers of AggressiveInlining.

Reply:
👍  removed locals